packages/graphql/tests/integration/issues/388.int.test.ts

```diff
@@ -55,7 +55,7 @@ describe("https://github.com/neo4j/graphql/issues/388", () => {
                         MATCH (myFriends)-[:HAS_POST]->(post:Post)
                         RETURN post
                     }
-                    RETURN DISTINCT apoc.map.merge(properties(post), { __resolveType: 'Post' }) AS result ORDER BY result.modifiedDate DESC
```

Comment:
This implementation uses `labels(<node>)` to resolve the node to a concrete type.  The `labels` function requires that it be passed a node, *not* a map  (the integration tests required this merging to be done in the cypher query as no attempt was being made to resolve it to a concrete type, but that is no longer the case).

That said, I can see that it may be useful to support custom cypher which returns a map corresponding to a union of types which do not have labels that exist in the database, so handling the case when a map could be useful in that scenario.

Is there a cypher function/utility to check if a variable is a map?

Then the code could be updated to be something like

`WHERE IS_MAP(this) OR ("U1" IN labels(this)) ... etc`

to skip calling labels, or attempting to further resolve fields, if the value returned is a map.

Reply:
I think how you have done it is fine, the hack users were using with __resolveType was a workaround and this is much better! On your question about function/utility to check if a variable is a map - I cant find any :/